tensorflow/contrib/go/tensor.go

```diff
@@ -0,0 +1,769 @@
+package tensorflow
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math"
+	"reflect"
+	"runtime"
+	"unsafe"
+
+	"github.com/golang/protobuf/proto"
+
+	pb "github.com/tensorflow/tensorflow/tensorflow/contrib/go/proto"
+)
+
+import "C"
+
+const (
+	cAckByte  = 6
+	cBellByte = 7
+	cDc1      = 17
+
+	cBytesComplex64 = 8
+	cBytesFloat32   = 4
+	cBytesFloat64   = 8
+	cBytesInt16     = 2
+	cBytesInt32     = 4
+	cBytesInt64     = 8
+	cBytesUint16    = 2
+)
+
+// A DataType represents the type of the data contained in a Tensor
+type DataType pb.DataType
+
+// A TensorInterface is the interface implemented by Tensors.
+type TensorInterface interface {
+	Data() []byte
+	DataSize() int64
+	DataType() DataType
+	GetVal(d ...int) (val interface{}, err error)
+
+	Dim(n int) int
+	NumDims() int
+
+	Bool() (res []bool, err error)
+	Float32() (res []float32, err error)
+	Float64() (res []float64, err error)
+	Int32() (res []int32, err error)
+	Int64() (res []int64, err error)
+	Str() (res [][]byte, err error)
```

Comment:
I'll change it by Byte since is what is really returning. Originally this methods was AsInt64, etc, but we decided to remove the As, I'm not 100% sure if this was a good idea.